### ISSUE TITLE
[accessibility issue] Improve accessibility for Dashboard VariableInput component

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79630]
**Issues**
1. [Relinking programmatic association issue.](https://github.com/grafana/grafana/issues/79517)
2. It is possible to submit a value which is not in the Select dropdown. In most (all?) cases this seems undesirable. The implementation should maybe more closely resemble our [Basic Select example in the documentation](https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--basic)?
3. Related to the above (but not necessarily a direct issue with the VariableInput component but related), if there is an error thrown on a variable input, no error message is provided to assistive technologies, such as screen readers.
4. It is unclear to users if the variable input is required or not.
5. Upon entering a value the keyboard focus is 'lost' and reset to the body element.
6. Upon opening the dropdown list, the expected behaviour is the first option is selected by default.
7. The input field does not have a role of `combobox`, it is unclear to users of assistive technologies that there are multiple options available.

**Steps to reproduce**
Go to any Grafana instance and find a dashboard panel with custom variables. 

UHG Audit IDs:
2.43
3.322
4.121
4.13

**Comment by danfoxley** [comment]
@ckbedwell excuse the barge in on this issue. Is this something to add to this issue, or should I make a separate?
I've added a Text box Variable (to a Loki data source) dashboard. I'm using it to search/filter a Loki query as a Line contains $search variable. Should the greyed out text prompt "Enter variable value" on the dashboard be adjustable or blank?  Using Text box for "search" doesn't seem to match the prompt.

Dashboard input box. The input box shows the greyed out text prompt: "Enter variable value"
<img width="328" alt="image" src="https://github.com/grafana/grafana/assets/2074038/d039fab1-8952-48cc-a922-f2ef278482ab">


Variable
<img width="342" alt="image" src="https://github.com/grafana/grafana/assets/2074038/4e2062b0-7126-4571-9a07-5862be00515e">

**Comment by ckbedwell** [comment]
@danfoxley I think it is worth making a new issue from what you are describing. This issue is fairly specific to one of our accessibility audits so it's probably easier to keep track of what you are describing separately. Maybe link to this issue in your new one so when the VariableInput gets worked on there's increased visibility for everything related to it.

**Comment by danfoxley** [comment]
> @danfoxley I think it is worth making a new issue from what you are describing.  I created a specific issue https://github.com/grafana/grafana/issues/80111

Should I delete my post, so as not to create noise here?

**Comment by tskarhed** [comment]
Some of my thoughts that might go into this, which requires a little bit more refactoring:
- Use `Input` component (I don't think we want to use Select here at the moment)
- Replace the fake button with an input field
- Don't switch between input fields when clicking/focusing, as it kinda ruins some a11y stuff

**Comment by tskarhed** [comment]
In Scenes dashboards, this component is a select, which means that we could just wait and it will be fixed automatically once dashboards are running Scenes.

**Comment by tskarhed** [comment]
We should revisit this issue with Combobox rolled out

**Comment by taleena** [comment]
Things have likely changed a bit since this was filed, so may need some reinvestigation. Add a tool to check this in development mode as the risk of this popping up again is high.

**Comment by taleena** [comment]
@ashharrison90 we should re-validate this, it looks like there have been some issues filed for the combobox, but it is not clear if those should replace this one.
